### PR TITLE
Return correct exit code with side conditions

### PR DESCRIPTION
**Pull request by ttuegel**
The correct exit code is returned when the final configuration has side conditions.

Fixes: #1872 

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

**Comment by andreiburdusa** [comment]
Sorry! I did a checkout to this branch so I could look at the code locally with my editor. Then I wanted to update my master branch but instead I updated yours. I hope I won't cause any problems

**Comment by ttuegel** [comment]
@andreiburdusa Don't worry, everything looks fine!

File: test/issue-1872/test.k [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module TEST
+  imports INT
+
+  configuration
+    <k> $PGM:Pgm </k>
+    <exitcode exit=""> 0 </exitcode>
+
+  syntax Pgm ::= "begin" | "even" | "odd"
+  rule begin => ?_:Int
```
Comment (ana-pantilie) [inline test/issue-1872/test.k:9]: Is it certain that the odd rule will be the first to apply here?

Comment (ttuegel) [reply test/issue-1872/test.k:9]: Thanks for reminding me! I changed the definition to remove the `even` branch. Now there is only one branch, but it still has a side condition.

**Pull request merged**